### PR TITLE
Cinema spring 04

### PR DESCRIPTION
**Pull request by Rudiachenko**

### CODE REVIEW

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -23,7 +22,7 @@
 public class Main {
     private static final Logger logger = Logger.getLogger(Main.class);
 
-    public static void main(String[] args) throws AuthenticationException {
+    public static void main(String[] args) {
```
Comment (andromash) [inline src/main/java/com/dev/cinema/Main.java]: I think you can delete this class already

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 2]
```diff
@@ -1,6 +1,9 @@
 package com.dev.cinema.model.dto;
 
+import javax.validation.constraints.NotNull;
+
 public class MovieSessionRequestDto {
+    @NotNull
     private String showTime;
     private Long cinemaHallId;
     private Long movieId;
```
Comment (andromash) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java:11]: show Time is not null but is it even possible to have a movie session with null cinema hall and movie?

**Review by andromash — COMMENTED** [review]
Good job!

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder managerBuilder) {
+        try {
+            managerBuilder.inMemoryAuthentication()
+                    .withUser("admin")
+                    .password(passwordEncoder().encode("admin"))
+                    .roles("USER");
+        } catch (Exception e) {
+            throw new RuntimeException("Can't configure global user", e);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: I think this is redundant

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                    .and()
+                    .formLogin()
+                    .permitAll()
+                    .and()
+                    .httpBasic()
+                    .and()
+                    .csrf().disable();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't authorize user", e);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: same

File: src/main/java/com/dev/cinema/controllers/UserController.java [hunk 5]
```diff
@@ -22,9 +23,9 @@ public UserController(UserService userService, UserMapper userMapper) {
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto get(@RequestParam String email) {
-        return userService.findByEmail(email)
-                .map(userMapper::toUserResponseDto)
-                .orElseThrow();
+    public UserResponseDto get(Authentication authentication) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/UserController.java]: let's not change this - this method should return info about any user by his email

File: src/main/java/com/dev/cinema/validation/EmailValueValidator.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValueValidator implements ConstraintValidator<EmailValueConstraint, String> {
+    private static final String EMAIL_REGEX = "^[\\w!#$%&'*+/=?`{|}~^-"
+            + "]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*"
+            + "@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email.matches(EMAIL_REGEX);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/validation/EmailValueValidator.java]: let's add null check - and we can remove @NotNull constraint above email field
the problem is that we can not ensure the order of validation

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**